File: frontend/templates/react_clean_architecture/src/gateway/dummy-gateway/src/dummy-gateway.ts
import type { TemplateItem } from "@template/model";
import type { ItemGateway } from "@template/usecase";

/** メモリ上の固定データを返す、開発・テスト用のGatewayです。 */
export class DummyGateway implements ItemGateway {
  readonly #items: readonly TemplateItem[];

  constructor(initialItems: readonly TemplateItem[] = []) {
    this.#items = [...initialItems];
  }

  async findAll(): Promise<readonly TemplateItem[]> {
    return [...this.#items];
  }
}

File: mobile/templates/riverpod_clean_architecture/lib/gateway/dummy_gateway/lib/dummy_gateway.dart
import 'package:model/model.dart';
import 'package:usecase/usecase.dart';

/// メモリ上の固定データを返す、開発・テスト用のGatewayです。
final class DummyGateway implements ItemGateway {
  DummyGateway({List<TemplateItem> initialItems = const <TemplateItem>[]})
    : _items = List<TemplateItem>.from(initialItems);

  final List<TemplateItem> _items;

  @override
  Future<List<TemplateItem>> findAll() async =>
      List<TemplateItem>.unmodifiable(_items);
}
